react/features/virtual-background/components/VirtualBackgroundDialog.js

```diff
@@ -54,23 +55,73 @@ type Props = {
  * @returns {ReactElement}
  */
 function VirtualBackground({ dispatch, t }: Props) {
+    const localImages = localStorage.getItem('storedImages');
+    const [ storedImages, setStoredImages ] = useState((localImages && JSON.parse(localImages)) || []);
+    const [ loading, isloading ] = useState(false);
+
+
+    /**
+     * Updates stored images on local storage.
+     */
+    useEffect(() => {
+        localStorage.setItem('storedImages', JSON.stringify(storedImages));
+    }, [ storedImages ]);
+
     const [ selected, setSelected ] = useState('');
-    const enableBlur = () => {
+    const enableBlur = async () => {
+        isloading(true);
         setSelected('blur');
-        dispatch(setVirtualBackground('', false));
-        dispatch(toggleBackgroundEffect(true));
+        await dispatch(setVirtualBackground('', false));
+        await dispatch(toggleBackgroundEffect(true));
+        isloading(false);
     };
 
-    const removeBackground = () => {
+    const removeBackground = async () => {
+        isloading(true);
         setSelected('none');
-        dispatch(setVirtualBackground('', false));
-        dispatch(toggleBackgroundEffect(false));
+        await dispatch(setVirtualBackground('', false));
+        await dispatch(toggleBackgroundEffect(false));
+        isloading(false);
     };
 
-    const addImageBackground = image => {
+    const addImageBackground = async image => {
+        isloading(true);
         setSelected(image.id);
-        dispatch(setVirtualBackground(image.src, true));
-        dispatch(toggleBackgroundEffect(true));
+        await dispatch(setVirtualBackground(image.src, true));
+        await dispatch(toggleBackgroundEffect(true));
+        isloading(false);
+    };
+
+    const deleteStoredImage = image => {
+        setStoredImages(storedImages.filter(item => item !== image));
+    };
+
+    const uploadImage = async imageFile => {
+        const reader = new FileReader();
+
+        reader.readAsDataURL(imageFile[0]);
```

Comment:
Does this mean it will only upload the first image if I select several?

Reply:
You can't select multiple images for upload. The file input is set by default to accept a single file. The multiple property is not set.  Related to '[0]' -> this is how the file structure appears by default when you have a single file for upload.